### DISCUSSION TITLE
Is `Object.groupBy` working?

### DISCUSSION THREAD

**Discussion by Semigradsky** [zloirock/core-js#1291] [Q&A] [answered]
Looks like `Object.groupBy` doesn't work with babel.

My configuration:
```
@babel/preset-env: 7.22.20
@babel/compat-data: 7.22.20
core-js-compat: 3.31.0
```

Babel config:
```
    [
      '@babel/preset-env',
      {
        useBuiltIns: 'usage',
        corejs: 3.32,
        bugfixes: true,
        shippedProposals: true,
      },
    ],
```

The code `console.log(Object.groupBy({}, ({ foo }) => foo))` fails

https://github.com/tc39/proposal-array-grouping on Stage 3

**Comment by zloirock** [comment, marked as answer]
`console.log(Object.groupBy({}, ({ foo }) => foo))` fails because you pass a non-iterable object.

Works something like that:

```js
console.log(Object.groupBy([{ foo: 1 }, { foo: 2 }], ({ foo }) => foo)); // => { 1: [{ foo: 1 }], 2: [{ foo: 2 }] }
```

**Reply by zloirock** [reply]
If it will not work for you, ping Babel team in the https://github.com/babel/babel-polyfills repo since theoretically the current published `babel-plugin-polyfill-corejs3` package version could use obsolete data and does not know that `Object.groupBy` is shipped.

Or just pass `proposals: true`.

**Reply by Semigradsky** [reply]
As I see `groupBy` exists in `polyfill-corejs3`:
https://github.com/babel/babel-polyfills/blob/8999c65c946789931fd68de3be45775dff0df334/packages/babel-plugin-polyfill-corejs3/src/shipped-proposals.ts#L27

Could you advise where I can found code for detection that need to add polyfill? For debugging it

**Reply by zloirock** [reply]
@Semigradsky as I wrote above, it seems [the latest published version of `babel-plugin-polyfill-corejs3` is 0.8.3 - it's July](https://www.npmjs.com/package/babel-plugin-polyfill-corejs3?activeTab=versions) and `Object.groupBy` still was not shipped.

https://github.com/babel/babel-polyfills/blob/babel-plugin-polyfill-corejs3%400.8.3/packages/babel-plugin-polyfill-corejs3/src/shipped-proposals.ts

I pushed changes with this after that, but it seems the new version of the package is still not published.

@nicolo-ribaudo it seems need to do it.

**Reply by Semigradsky** [reply]
Oh, I see. Thank you
